docs/DeveloperGuide.md

```diff
@@ -1,38 +1,253 @@
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+listEntry()
 
-## Non-Functional Requirements
+* Takes in a list of Entry and passes them to the UI to print to the user.
+* Makes use of toString() method of entries.
+
+deleteEntry()
+
+* Takes in the list of entry and index of the entry to be deleted.
+* The method acts as an underlying method for subclasses' deletion methods.
+
+editEntry()
+
+* Overloaded method. All methods take in the LinkedList of entries and the index of the entry that has to be edited.
+  Another parameter has to be provided, which can be double, LocalDateTime, or String.
+* additional parameter double edits the money amount of the entry in the list.
+* additional parameter LocalDateTime edits the time when the entry is made.
+* additional parameter String edits the description of the entry in the list.
+
+getTotalAmount()
+
+* Takes in a list of entries and returns the sum of all entry's monetary value.
+
+SortEntryByAmount()
+
+* Returns a sorted list of entry by amount of entry.
+
+SortEntryByDate()
+
+* Returns a sorted list of entry by date of entry.
+  
+findEntriesByCategory()
+* Returns a list of entries with the same category and the total amount in this category.
+
+### ExpenseList, IncomeList
+
+Classes ExpenseList and IncomeList is responsible for keeping track of the corresponding entry instances added to the program by the user. 
+At a class level, it keeps a **LinkedList** of corresponding entries. 
+They both extend EntryList, the abstract class that represents a collection of Entry instances. 
+It provides static functionalities of managing and viewing entry instances at a class level. 
+Instances of ExpenseList and IncomeList are not created as all functionalities can be provided at the class level. 
+  
+**Methods**
+
+listExpense() / listIncome()
+
+* Overloaded method, may take it no parameter or LocalDate parameter
+* If it has no passed parameters, it returns all entries in the list
+* If a LocalDate is passed, it returns all entries made in the month specified by LocalDate instance.
+
+### Budget
+
+The Budget class represents the user’s monthly budget. The class utilize class-level hashmaps to represent the monthly
+budget, using outer key year and inner key month. It provides a method to set and view budget for different months.
+
+It makes use of a static HashMap<Integer, HashMap<Month, Double>> to keep track of monthly budget. If the user tries to
+access budget using keys that are not entered in the HashMap, it will return a warning mentioning that the inquired
+budget has not been set yet.
+
+### SaveExpense, SaveIncome
+
+The SaveExpense and SaveIncome class deal with saving in the user inputted data locally so that it can be later accessed.
+**Methods**
```

Comment:
![image](https://user-images.githubusercontent.com/88487773/228706632-3fcf6a6e-b17f-44bf-8376-6fcd7fcfbb32.png)
Can consider adding a line break here so that "Methods" is on a new line